### PR TITLE
[exporter/elasticsearch] Add summary support for metrics

### PR DESCRIPTION
**Pull request by carsonip**
**Description:** <Describe what has changed.>

Add summary support for metrics

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>
Added exporter test

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/elasticsearchexporter/model.go [hunk 1]
```diff
@@ -321,6 +321,15 @@ func (m *encodeModel) upsertMetricDataPointValue(documents map[uint32]objmodel.D
 	return nil
 }
 
+func summaryToValue(dp pmetric.SummaryDataPoint) (pcommon.Value, error) {
+	// quantiles are ignored
```
Comment (lahsivjar) [inline exporter/elasticsearchexporter/model.go:325]:
```suggestion
	// TODO: Add support for quantiles
```
Also, can we create an issue for this?

File: exporter/elasticsearchexporter/exporter.go [hunk 2]
```diff
@@ -265,6 +265,20 @@ func (e *elasticsearchExporter) pushMetricsData(
...
 					}
+				case pmetric.MetricTypeSummary:
+					dps := metric.Summary().DataPoints()
+					for l := 0; l < dps.Len(); l++ {
+						dp := dps.At(l)
+						val, err := summaryToValue(dp)
+						if err != nil {
+							errs = append(errs, err)
+							continue
+						}
```
Comment (lahsivjar) [inline exporter/elasticsearchexporter/exporter.go:276]: _nit_: I don't think this would ever return an error, maybe we can simplify

**Review by lahsivjar — APPROVED** [review]
LGTM! A few minor comments.

File: exporter/elasticsearchexporter/model.go (hunk 1 above)
Comment (carsonip) [reply exporter/elasticsearchexporter/model.go]: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34561

**Review by MovieStoreGuy — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
Looks like a couple of checks are failing:

```
Error: elasticsearchexporter/utils_test.go:39:69: `assertItemsEqual` - `assertOrder` always receives `false` (unparam)
level=info msg="[runner] linters took 16.095711067s with stages: goanalysis_metalinter: 16.088054729s"
level=info msg="File cache stats: 24 entries of total size 218.1KiB"
func assertItemsEqual(t *testing.T, expected, actual []itemRequest, assertOrder bool) {
```

**Comment by carsonip** [comment]
@codeboten It should be fixed in an earlier PR #34471. Merge main and rerunning checks now.

**Review by codeboten — APPROVED** [review]

**Pull request merged**